### PR TITLE
Add runtime support for deserializing HeaderCollections

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

File: client-runtime/src/main/java/com/microsoft/rest/v2/protocol/HttpResponseDecoder.java [hunk 1]
```diff
@@ -292,6 +296,50 @@ private Object deserializeHeaders(HttpHeaders headers) throws IOException {
...
+                                            headerCollection.put(headerName.substring(headerCollectionPrefixLength), header.value());
+                                        }
+                                    }
+
+                                    final boolean declaredFieldAccessibleBackup = declaredField.isAccessible();
+                                    try {
+                                        if (!declaredFieldAccessibleBackup) {
+                                            declaredField.setAccessible(true);
+                                        }
+                                        declaredField.set(deserializedHeaders, headerCollection);
```
Comment (RikkiGibson) [inline client-runtime/src/main/java/com/microsoft/rest/v2/protocol/HttpResponseDecoder.java]: Ah, so when we serialize/deserialize we take care of all the fixed-size fields? And then we use reflection to set the HeaderCollection property?

Comment (daschult) [reply client-runtime/src/main/java/com/microsoft/rest/v2/protocol/HttpResponseDecoder.java]: Correct.

Comment (daschult) [reply client-runtime/src/main/java/com/microsoft/rest/v2/protocol/HttpResponseDecoder.java]: I'll add the HeaderCollection annotation in the generator.

**Comment by RikkiGibson** [comment]
Just Checkstyle failures

**Pull request merged**